src/storage/download.rs

```diff
@@ -94,21 +93,41 @@ impl<T: ?Sized> CachedBackend for T
 where
     T: Backend,
 {
+    /// Get the latest version of a component across all supported environments
```

Comment:
Add the note from #7 about how this returns a version which is available in all (supported) environments.
Even though a higher version is available, if it isn't everywhere, it won't be chosen.